pkg/gather/clusterconfig/clusterconfig.go

```diff
@@ -36,7 +36,7 @@ func New(client configv1client.ConfigV1Interface) *Gatherer {
 var reInvalidUIDCharacter = regexp.MustCompile(`[^a-z0-9\-]`)
 
 func (i *Gatherer) Gather(ctx context.Context, recorder record.Interface) error {
```

Comment:
The main issue I have with the current state is that we gave two independent loops doing part of the process, that already proved to be error-prone (getting the gathered data from the same time split between two archives, or even loosing them in some corner cases).

I don't insist about the config gatherer calling the `Flush` (works mainly when there is only one Gatherer). So I would propose having 'GatheringLoop` object, that would be THE looping place, calling gathereres first and flushing once they finish. This object can also be responsible with dealing with long-running gatherers, timeouts etc.

After that change, neither gatherers or recorders would have their own loop, but they would be instrumented from a single place owning the life cycle..

Reply:
#14 and #15 add more data - I think we want ctx in those anyway.  I think I can live with this change for now and we can always revisit.

Let me get some results returned from those to verify - they change the structure of this method slightly

Reply:
Actually, why not have the person calling gather do the flush?   I don't think the gatherer needs to know, but the thing calling it can easily do this.